javascript-carousel/main.js

```diff
@@ -0,0 +1,82 @@
+var $view = document.querySelector('.container');
+var $viewNumber = $view.getAttribute('view');
+var $imageSource = document.querySelector('img');
+var $leftAngle = document.querySelector('.fa-chevron-left');
+var $rightAngle = document.querySelector('.fa-chevron-right');
+var $dot = document.querySelectorAll('.dot');
+var timer = setInterval(autoRun, 3000);
+
+var images = [
+    {
+      imageID: 1,
+      source: "images/001.png"
+    },
+    {
+      imageID: 2,
+      source: "images/004.png"
+    },
+    {
+      imageID: 3,
+      source: "images/007.png"
+    },
+    {
+      imageID: 4,
+      source: "images/025.png"
+    },
+    {
+      imageID: 5,
+      source: "images/039.png"
+    },
+  ]
+
+function switchImage() {
+  $view.setAttribute('view', $viewNumber);
+
+  for (let i = 0; i < images.length; i++) {
+    if ($viewNumber === images[i].imageID) {
+      $imageSource.setAttribute('src', images[i].source)
+    }
+  };
+
+  for (let j = 0; j < $dot.length; j++) {
+    if ($viewNumber.toString() === $dot[j].getAttribute('index')) {
+      $dot[j].setAttribute('class', 'dot fas fa-circle')
+    } else {
+      $dot[j].setAttribute('class', 'dot far fa-circle')
+    }
+  }
+}
+
+function autoRun() {
+  if ($viewNumber === 5) {
```

Comment:
I'd suggest basing these off of the number of images, rather than hard-coding them.

Reply:
do you mean it's better to use the length property instead of using a fixed number, cause there might be situations that i don't know the total number of images?

Reply:
> do you mean it's better to use the length property instead of using a fixed number

@VeraYeh Yes.

> cause there might be situations that i don't know the total number of images?

Either that, or when you change the number of images, you only need to update it in one place.

Reply:
@TimelikeClosure 
Updated the if statements, replacing fixed number with length property to check the number of images.
![JavaScript Carousel - Profile 1 - Microsoft_ Edge 2021-06-29 20-29-21](https://user-images.githubusercontent.com/76827487/123898073-7b3df580-d919-11eb-93d6-8e4a1a1e1b12.gif)